tests/func/test_run_single_stage.py

```diff
@@ -1017,3 +1018,32 @@ def test_metrics_dir_not_cached(self, dvc):
                 metrics_no_cache=["dir"],
                 single_stage=True,
             )
+
+
+def test_run_force_doesnot_preserve_comments_and_meta(tmp_dir, dvc, run_copy):
```

Comment:
Took a lot of time to find out why this was not the case. It's simple though: we never load `stage_text` on `run` and hence, when `dump`-ing, `to_single_stagefile` has no idea of `meta` or comments. Anyway, the PR is focused on order for `dvc.yaml`. I just added this test, if someone gets confused later.